.travis.yml

```diff
@@ -43,58 +43,46 @@ services: postgresql
 sudo: required
 
 install:
-  - composer install --working-dir=src
+  - ./utils/prepare-test -afty
+  - composer install --prefer-dist --working-dir=src
   - ./install/scripts/install-spdx-tools.sh
   - ./install/scripts/install-ninka.sh
 
-before_script:
-  - ./utils/prepare-test -afty
-
-env:
-  global:
-    - PHPTESTSUITE=''
-    - CHECKBEFORE='cppcheck -q -isrc/nomos/agent_tests/testdata/NomosTestfiles/ -isrc/testing/dataFiles/ --suppress=*:src/copyright/agent/json.hpp src/'
-    - MAKETARGETS='all test'
-  matrix:
-    - PHPTESTSUITE='Fossology PhpUnit Test Suite' CHECKBEFORE='' MAKETARGETS='build-lib VERSIONFILE build-cli'
-
 script:
-  - set -e
   - src/testing/syntax/syntaxtest.sh
-  - if [[ ${TRAVIS_PHP_VERSION:0:3} == "5.6" ]]; then composer require --no-update phpunit/phpunit ^5; fi
-  - if [[ ${TRAVIS_PHP_VERSION:0:3} == "7.0" ]]; then composer require --dev --no-update phpunit/phpunit ^6; fi
-  - if [[ ${TRAVIS_PHP_VERSION:0:3} == "7.1" ]]; then composer require --dev --no-update phpunit/phpunit ^7; fi
-  - if [[ ! -z "${CHECKBEFORE}" ]]; then ${CHECKBEFORE}; fi
-  - if [[ ! -z "${MAKETARGETS}" ]]; then make ${MAKETARGETS}; fi
-  - if [[ ! -z "${PHPTESTSUITE}" ]]; then src/vendor/bin/phpunit -csrc/phpunit.xml --testsuite="${PHPTESTSUITE}"; fi
-  - set +e
+  - make build-lib VERSIONFILE build-cli
+  - src/vendor/bin/phpunit -csrc/phpunit.xml --testsuite="Fossology PhpUnit Test Suite"
 
 after_script:
   - mkdir -p build/logs
-  - if [[ ! -z "${PHPTESTSUITE}" ]]; then php src/vendor/bin/coveralls -vv -x clover.xml; fi
+  - php src/vendor/bin/coveralls -vv -x clover.xml
 
 matrix:
   include:
-  ## docker tests ##
-    - name: default-docker-test
-      services: docker
-      install: true
-      before_script: true
-      script: src/testing/docker/default-docker-test.sh
-    - name: docker-compose-test
+    - name: Docker Tests
+      addons: {}
       services: docker
-      install: true
-      before_script: true
-      script: src/testing/docker/docker-compose-test.sh
+      install: skip
+      script:
+        - docker-compose build
+        - src/testing/docker/test-cluster.sh
+        - src/testing/docker/test-standalone.sh
+      after_script: skip
   ## C/C++ agent tests ##
-    - env: CC=gcc-4.8 CXX=g++-4.8 CFLAGS='-Wall'
+    - &compiler-tests
+      env: CC=gcc-4.8 CXX=g++-4.8 CFLAGS='-Wall'
       addons:
         apt:
           packages:
```

Comment:
Consider adding `sources: ubuntu-toolchain-r-test` for gcc-4.8 as well as it may break tests in future Travis versions.

Reply:
Done.